src/Numerics/Mesh/FTP_kernels.jl

```diff
@@ -0,0 +1,379 @@
+# This kernel computes the fast tensor-product x vector
+# vout = (phit ⊗ phis ⊗ phir) * vin
+# using Julia native multi-threading
+
+function ftpxv_hex_CPU!(
+    vin::AbstractArray{FT, 2},
+    vin_den::Union{AbstractArray{FT, 2}, Nothing},
+    vout::AbstractArray{FT, 2},
+    phir::Union{AbstractArray{FT, 2}, Nothing},
+    phis::Union{AbstractArray{FT, 2}, Nothing},
+    phit::Union{AbstractArray{FT, 2}, Nothing},
+    si::Int,
+    sj::Int,
+    sk::Int,
+    sr::Int,
+    ss::Int,
+    st::Int,
+    temp::AbstractArray{FT, 3},
+) where {FT <: AbstractFloat}
+    Nel = size(vin, 2)
+    Threads.@threads for e in 1:Nel
+        #--------------------------------------------------------------------------------
+        if !(phir === nothing) # apply phir
```

Comment:
```suggestion
        if phir !== nothing # apply phir
```

Reply:
I've always thought there should be an `issomething` to pair with `isnothing`